src/core/QCheck2.ml

```diff
@@ -1099,13 +934,15 @@ end = struct
           with Not_found ->
             if extend then (
               (* Generate a new value and "record" the binding for potential future display/shrinking *)
-              let value_tree = v_arb.gen st in
+              let value_tree = v_gen st in
               p_tree_bindings_rev := (key, value_tree) :: !p_tree_bindings_rev;
               let v = Tree.root value_tree in
               T.add tbl key v;
               Some v
             ) else None) in
```

Comment:
maybe the whole `create` could take an optional printer? Or even more modular, the function to print a table could take a printer for the key and value types, as it is done in the stdlib and elsewhere.

Reply:
I prefer the optional printer in `create` as it is used in `print_repr`
```ocaml

  (** [print_rep repr] returns a string representation of [repr]. *)
  let print_repr r =
    let buf = Buffer.create 32 in
    let rec aux
      : type f. Buffer.t -> f fun_repr -> unit
      = fun buf r -> match r with
        | Fun_map (_, sub_repr) -> aux buf sub_repr
        | Fun_tbl r ->
           Buffer.add_string buf (Poly_tbl.print r.fun_tbl);  (* <-- here *)
           Printf.bprintf buf "_ -> %s" (match r.fun_arb.print with
              | None -> "<opaque>"
              | Some print -> print r.fun_default
            ); *)
    in
    Printf.bprintf buf "{";
    aux buf r;
    Printf.bprintf buf "}";
    Buffer.contents buf
```
But it's not a module that I use, don't really know what's the most suitable. It depends on if
`Poly_tbl.print` is mostly used throught `print_repr` or by itself.

Reply:
I put the printer as an optional printer in those functions.

Reply:
I did put optional printer at various places, see  https://github.com/c-cube/qcheck/pull/131/commits/05181aa1552aecb1f94af2800c5f382c9fbd7ceb.